### ISSUE TITLE
Watcher for CRD timeout in 90sec with EOF

### ISSUE THREAD

**Issue by hardwaysun** [ericchiang/k8s#116]
Not sure it is K8s doing it or something in the package. When watch  resource like validatingwebhookconfiguration, it return from the watch.Next with EOF every 40 min , but if I define crd and watch it, it return every 90 second, tried adjust http.Transport{IdleConnTimeout} or even the --min-request-timeout=1800 on apiserver, all doesn't help.

Intuitively it should be same as it is just a resource on apiserver we doing long poll, should not matter what we looking for, but struggled for 3 days, can't find the issue.  See log below, mytestcrd return very 90sec, whild the validatingwebhoodconfiguration do so around 45min

2019-04-09T12:32:16.168|ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=validatingwebhookconfiguration
........................
2019-04-09T13:21:58.73 |ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=mytestcrds
2019-04-09T13:22:21.342|ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=validatingwebhookconfiguration
2019-04-09T13:23:22.714|ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=mytestcrds

**Comment by silenceli** [comment]
I meet the some problem. I don't known how to solve

**Comment by silenceli** [comment]
> Not sure it is K8s doing it or something in the package. When watch resource like validatingwebhookconfiguration, it return from the watch.Next with EOF every 40 min , but if I define crd and watch it, it return every 90 second, tried adjust http.Transport{IdleConnTimeout} or even the --min-request-timeout=1800 on apiserver, all doesn't help.
> 
> Intuitively it should be same as it is just a resource on apiserver we doing long poll, should not matter what we looking for, but struggled for 3 days, can't find the issue. See log below, mytestcrd return very 90sec, whild the validatingwebhoodconfiguration do so around 45min
> 
> 2019-04-09T12:32:16.168|ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=validatingwebhookconfiguration
> ........................
> 2019-04-09T13:21:58.73 |ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=mytestcrds
> 2019-04-09T13:22:21.342|ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=validatingwebhookconfiguration
> 2019-04-09T13:23:22.714|ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=mytestcrds

Do you find the answer now?

**Comment by hardwaysun** [comment]
Nope. I end up only use the validatingwebhookconfigration to get my crd, can't directly watch it.
    On Thursday, March 19, 2020, 03:00:27 AM PDT, silenceli <notifications@github.com> wrote:  
 
 



Not sure it is K8s doing it or something in the package. When watch resource like validatingwebhookconfiguration, it return from the watch.Next with EOF every 40 min , but if I define crd and watch it, it return every 90 second, tried adjust http.Transport{IdleConnTimeout} or even the --min-request-timeout=1800 on apiserver, all doesn't help.

Intuitively it should be same as it is just a resource on apiserver we doing long poll, should not matter what we looking for, but struggled for 3 days, can't find the issue. See log below, mytestcrd return very 90sec, whild the validatingwebhoodconfiguration do so around 45min

2019-04-09T12:32:16.168|ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=validatingwebhookconfiguration
........................
2019-04-09T13:21:58.73 |ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=mytestcrds
2019-04-09T13:22:21.342|ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=validatingwebhookconfiguration
2019-04-09T13:23:22.714|ERRO|CTL|resource.(*kubernetes).startWatchResource.func1: Watch failure - error=decode event: EOF resource=mytestcrds


Do you find the answer now?
